Create valgrind.adoc (#352)
freebsd/freebsd-doc: website/content/en/status/report-2024-01-2024-03/valgrind.adoc

Contact: Paul Floyd <pjfloyd@wanadoo.fr>

The major news, as per the title, is that a port to FreeBSD arm64 (or aarch64) is now ready.
The next steps are to get it reviewed and pushed upstream.

Valgrind 3.23 due out at the end of April 2024 and package:devel/valgrind[] will be updated shortly after that.

package:devel/valgrind-devel[] will get an update as soon as I've pushed the changes for arm64.

`--track-fds=yes` now checks for and warns about attempts to close a file decriptor more than once.
Handling of closefrom has been improved to use this feature.

There are some important fixes for FreeBSD 15, in particular handling the new `libsys`.

smaller bugfixes.

Support for FreeBSD 13.3 has been added.
Added a redirect for `reallocarray`.
Several fixes for `aio*` functions.
Added a redirect for `memccpy`.
There is a fix for `_umtx_op OP_ROBUST_LISTS`.
Added redirs for C23 `free_sized` and `free_aligned_sized`.
Correctly propagate the ELF stack protection flags to the guest stack that Valgrind synthesizes.
Fixes for `--sanity-level-3` and above (only used for Valgrind self-testing at runtime).

> Aeveral fixes to checking done for `semctl`.

Reviewer: s/Aeveral/Several/

> Fixed argument checking for `utrace`.

Author: That would work.

=== Valgrind: port to arm64 on its way

Links: +
link:https://www.valgrind.org/[Valgrind Home Page] URL: link:https://www.valgrind.org/[] +
link:https://www.valgrind.org/docs/manual/dist.news.html[Valgrind News] URL: link:https://www.valgrind.org/docs/manual/dist.news.html[]
link:https://github.com/paulfloyd/freebsdarm64_valgrind[arm64 port] URL: link:https://github.com/paulfloyd/freebsdarm64_valgrind[]

Contact: Paul Floyd <pjfloyd@wanadoo.fr>

The major news, as per the title, is that a port to FreeBSD arm64 (or aarch64) is now ready.
The next steps are to get it reviewed and pushed upstream.

> Valgrind 3.23 due out at the end of April 2024 and package:devel/valgrind[] will be updated shortly after that.

Reviewer 2: Mind that I am not a native English speaker, but this does not sound right to me. Maybe
```suggestion
Valgrind 3.23 is due out at the end of April 2024 and package:devel/valgrind[] will be updated shortly after that.
```

Author: OK

> package:devel/valgrind-devel[] will get an update as soon as I've pushed the changes for arm64.

Reviewer 2: Please avoid contractions.
```suggestion
package:devel/valgrind-devel[] will get an update as soon as I have pushed the changes for arm64.
```

Author: OK as well

`--track-fds=yes` now checks for and warns about attempts to close a file decriptor more than once.
Handling of closefrom has been improved to use this feature.

There are some important fixes for FreeBSD 15, in particular handling the new `libsys`.

Here is a list of smaller bugfixes:

* Support for FreeBSD 13.3 has been added.
* Added a redirect for `reallocarray`.
* Several fixes for `aio*` functions.
* Added a redirect for `memccpy`.
* There is a fix for `_umtx_op OP_ROBUST_LISTS`.
* Added redirs for C23 `free_sized` and `free_aligned_sized`.

> * Correctly propagate the ELF stack protection flags to the guest stack that Valgrind synthesizes.

Author: No, it should be present continuous, this is something that gets done at every startup. Previously this was hard coded and did not respect the guest stack protection flags.

There are some important fixes for FreeBSD 15, in particular handling the new `libsys`.

Here is a list of smaller bugfixes:

* Support for FreeBSD 13.3 has been added.
* Added a redirect for `reallocarray`.
* Several fixes for `aio*` functions.
* Added a redirect for `memccpy`.
* There is a fix for `_umtx_op OP_ROBUST_LISTS`.

> * Added redirs for C23 `free_sized` and `free_aligned_sized`.

Reviewer: ```suggestion
* Added redirects for C23 `free_sized` and `free_aligned_sized`.
```

Author: OK
